test/test_importing.py

```diff
@@ -1,31 +1,20 @@
-import pytest as pytest_collecting
-
 from .support import HPyTest
 
-# this function should probably goes somewhere into hpy.universal and/or and
-# hpy package and/or an import hook, or whatever. I do not want to think about
-# this now.
-def import_module_properly(mod):
-    raise NotImplementedError("fix me eventually")
-
-# this was moved from support.py, where it did not belong
-## class HPyLoader(ExtensionFileLoader):
-##     def create_module(self, spec):
-##         import hpy.universal
-##         return hpy.universal.load_from_spec(spec)
-
 
 class TestImporting(HPyTest):
 
-    @pytest_collecting.mark.xfail
     def test_importing_attributes(self):
+        import pytest
         import sys
-        modname = 'mytest'
-        so_filename = self.compile_module("""
+        if not hasattr(sys, "executable"):
```

Comment:
Is is to check whether we are inside app-level tests or not. I should probably factor this out into a small helper function but I'm not particular sure what to call it. `is_app_level` is one idea, but perhaps that's too PyPy specific?

Reply:
ideally, the HPy tests should be independent of the implementation. But practically speaking, being able to do `if is_pypy_apptest` will probably be very useful.
What about introducing a generic way to check the implementation? E.g., we could have `self.implementation` which by default is `sys.implementation.name` (so, `'cpython'`, `'pypy'`, etc.), but pypy's `test/support.py` could override it and set to `'pypy_apptest'`. And we could also add a `self.is_pypy` which does `return self.implementation in ('pypy', 'pypy_apptest')` maybe?

Reply:
I added some more methods that describe the possible capabilities of the test runner that the hpy tests currently care about. I dabbled with making something more generic (e.g. a `.capabilities(...)` method that returns a set of supported behaviours) but that was actually more code and less clear than the simple methods, so I went back to those.